panel/widgets/chatbox.py

```diff
@@ -0,0 +1,560 @@
+        )
+        self._scroll_button.js_on_click(
+            code="""
+            const outerContainer = document.querySelectorAll(".bk-Column")[0].shadowRoot
+            const column = outerContainer.querySelector(".bk-Column")
+            column.scrollTop = -column.scrollHeight
+            """,
+            args={"chat_log": self._chat_log},
+        )
+
+        box_objects = [self._chat_title] if self.name else []
+        box_objects.extend([self._chat_log, self._scroll_button])
+        if self.allow_input:
+            self._prepend_input(box_objects, layout)
+        else:
+            self._message_inputs = {}
+            self._send_button = None
+        self._composite[:] = box_objects
+
+        # add interactivity
+        self.param.watch(self._refresh_log, "value")
+        self.param.trigger("value")
+
+    def _prepend_input(self, box_objects: List, layout: Dict[str, str]) -> None:
+        """
+        Append the input widgets to the chat box.
+        """
+        self._message_inputs = {}
+        for message_input_widget in self.message_input_widgets:
+            key = message_input_widget.name or message_input_widget.__class__.__name__
+            if isinstance(message_input_widget, type):  # check if instantiated
+                message_input_widget = message_input_widget()
+            self._message_inputs[key] = message_input_widget
+
+        self._send_button = Button(
+            name="Send",
+            button_type="default",
+            sizing_mode="stretch_both",
+            max_width=100,
+        )
+        for message_input in self._message_inputs.values():
+            if isinstance(message_input, TextInput):
+                # for longer form messages, like TextArea / Ace, don't
+                # submit when clicking away; only if they manually click
+                # the send button
+                message_input.param.watch(self._enter_message, "value")
+            message_input.sizing_mode = "stretch_width"
+        self._send_button.on_click(self._enter_message)
+
+        row_layout = layout.copy()
+        row_layout.pop("width", None)
+        row_layout.pop("height", None)
+        row_layout["sizing_mode"] = "stretch_width"
+
+        input_items = Tabs()
+        for message_input_name, message_input in self._message_inputs.items():
+            input_items.append(
+                (
+                    message_input_name,
+                    Row(message_input, self._send_button.clone(), **row_layout),
+                )
+            )
+
+        if len(self._message_inputs) == 1:
+            input_items = input_items[0]  # if only a single input, don't use tabs
+        box_objects.insert(0, input_items)
+
+    def _generate_pastel_color(self, string: str) -> str:
+        """
+        Generate a random pastel color in hexadecimal format.
+        """
+        seed = sum([ord(c) for c in string])
+        random.seed(seed)
+
+        rgb_range = (150, 185)
```

Comment:
Maybe have this as a parameter? It could be nice to customize this range.

Reply:
That's a good suggestion!